stl/inc/utility

```diff
@@ -644,6 +645,79 @@ template <size_t _Idx>
 inline constexpr in_place_index_t<_Idx> in_place_index{};
 #endif // _HAS_CXX17
 
+#if _HAS_CXX20
+template <class _Ty>
+inline constexpr bool _Is_integral_not_bool_char = is_integral_v<_Ty>
+    && !_Is_any_of_v<_Ty, bool, char,
+#ifdef _NATIVE_WCHAR_T_DEFINED
+        wchar_t,
+#endif // _NATIVE_WCHAR_T_DEFINED
+#ifdef __cpp_char8_t
+        char8_t,
+#endif // __cpp_char8_t
+        char16_t, char32_t>;
+
+// SAFE INTEGRAL COMPARISONS

```

Comment:
This is not a conventional SHOUTY BANNER. Per our shouty banner convention this should be `// FUNCTION TEMPLATE cmp_equal` (and similarly for the other function templates). (Let the record show that I hate the shouty banners, but it's still the convention until we can resolve #306 in the name of all that's right and good.)